File: 프로그래머스/1/42748. K번째수/K번째수.swift
import Foundation

func solution(_ array:[Int], _ commands:[[Int]]) -> [Int] {
    
    return commands.map { arr in
        let start = arr[0] - 1
        let end = arr[1] - 1
        let index = arr[2] - 1
        
        return array[start...end].sorted()[index]
    }
}

File: 프로그래머스/1/42748. K번째수/K번째수.cpp
#include <algorithm>
#include <iostream>
#include <vector>

using namespace std;

vector<int> solution(vector<int> array, vector<vector<int>> commands) {
    vector<int> answer;

    for (vector<int> command : commands) {
        int start = command[0] - 1, end = command[1] - 1, k = command[2] - 1;
        vector<int> v;
        
        for (int i = start; i <= end; i++) {
            v.push_back(array[i]);
        }

        sort(v.begin(), v.end());

        answer.push_back(v[k]);

    }

    return answer;
}